pkg/apis/core/validation/shoot.go

```diff
@@ -327,6 +328,13 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *core.ShootSpec, newObjectMeta met
 
 	allErrs = append(allErrs, validateNetworkingUpdate(newSpec.Networking, oldSpec.Networking, fldPath.Child("networking"))...)
 
+	if !reflect.DeepEqual(oldSpec.SchedulerName, newSpec.SchedulerName) {
+		// only allow to set an empty scheduler name to the default scheduler
+		if oldSpec.SchedulerName != nil || pointer.StringDeref(newSpec.SchedulerName, "") != v1beta1constants.DefaultSchedulerName {
```

Comment:
With defaulting in place, will the `oldSpec.SchedulerName` be ever set to nil ?
Moreover, why we are restricting users to only set `default scheduler`?

Reply:
The field of existing shoots it `nil`. The idea was to set the `schedulerName` field for these.